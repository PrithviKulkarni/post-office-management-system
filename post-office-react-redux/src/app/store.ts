import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit';
import letterReducer from '../features/LetterSlice';
import parcelReducer from '../features/ParcelSlice';
import depotReducer from '../features/DepotSlice';
import driverReducer from '../features/DriverSlice' 
export const store = configureStore({
  reducer: {
    letter: letterReducer,
    parcel: parcelReducer,
    depot: depotReducer,
    driver: driverReducer
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
