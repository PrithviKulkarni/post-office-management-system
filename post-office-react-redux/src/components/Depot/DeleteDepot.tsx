import { useState, SyntheticEvent } from 'react'
import { useParams, useNavigate } from 'react-router-dom'

import { Container, Button, Form } from 'react-bootstrap'

import { defaultDepot, Depot } from '../interfaces/depotinterface'

import { useAppDispatch } from '../../app/hooks'
import { deleteADepot } from '../../features/DepotSlice'

const DeleteDepot = () => {
  const dispatch = useAppDispatch()
  const navigate = useNavigate()

  const [errorMessage, setErrorMessage] = useState<string>('')

  let { depId } = useParams()
  let depotId: string = '0'
  if (!depId) {
    depId = '0'
  }
  depotId = depId
  console.log({ depotId })

  const submitHandler = (event: SyntheticEvent) => {
    event.preventDefault()
    const validated = [depId].every(Boolean)
    try {
      if (validated) {
        const updDepotData: Depot = defaultDepot

        if (depId) {
          updDepotData.id = depId
        }

        console.log({ updDepotData })

        dispatch(deleteADepot(updDepotData))

        navigate('/depots')
        window.location.reload()
      }
    } catch (error) {
      setErrorMessage('Depot NOT FOUND.')
      console.log('errorMessage: ', errorMessage)
    }
  }

  return (
    <Container>
      <Form
        onSubmit={submitHandler}
        className='deleteDepot'
        data-testid='deleteDepot'
      >
        <Button type='submit' className='button' data-testid='button'>
          Delete Depot
        </Button>
      </Form>
    </Container>
  )
}

export default DeleteDepot
