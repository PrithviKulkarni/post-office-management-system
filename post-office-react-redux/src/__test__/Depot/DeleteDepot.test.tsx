import { fireEvent, render, screen } from "@testing-library/react";
import { store } from "../../app/store";
import { Provider } from "react-redux";
import DeleteDepot from "../../components/Depot/DeleteDepot";
import { BrowserRouter } from "react-router-dom";

// afterEach(cleanup);
it("renders Delete depot correctly", () => {
  const { asFragment } = render(
    <Provider store={store}>
      <BrowserRouter>
        <DeleteDepot />
      </BrowserRouter>
    </Provider>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("Verifies the Delete depot render", () => {
  const { getByTestId } = render(
    <Provider store={store}>
      <BrowserRouter>
        <DeleteDepot />
      </BrowserRouter>
    </Provider>
  );

  const button = getByTestId('button')
  expect(button.textContent).toBe('Delete Depot')
  fireEvent.click(button)
  expect(button).toBeTruthy()
  expect(screen.getByTestId("deleteDepot")).toHaveClass("deleteDepot");
});