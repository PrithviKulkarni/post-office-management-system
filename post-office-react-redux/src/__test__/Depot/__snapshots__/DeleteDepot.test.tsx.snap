// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Delete depot correctly 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <form
      class="deleteDepot"
      data-testid="deleteDepot"
    >
      <button
        class="button btn btn-primary"
        data-testid="button"
        type="submit"
      >
        Delete Depot
      </button>
    </form>
  </div>
</DocumentFragment>
`;
